### PR TITLE
Updated with valve generalization, sensor data retrieval, and all valve and solenoid arrays populated

### PR DESCRIPTION
**Pull request by vainaviv**

### CODE REVIEW

File: Brain/Brain.ino [hunk 1]
```diff
@@ -40,9 +39,7 @@ sensorInfo all_ids[11] = {
    {"Load Cell Engine Right",     -1, -1, 10, 5, NULL}
 };
 
-sensorInfo sensor {"", 0, 0, 0, 0, 0};
-
-int numSensors = 11;
```
Comment (fazerlicourice7) [inline Brain/Brain.ino:45]: we want this var

File: Brain/brain_utils.h [hunk 2]
```diff
@@ -126,12 +123,24 @@ void take_action(valveInfo *valve, int action) {
...
+ * action in solenoids.h
+ */
+void take_action(valveInfo *valve, int action) {
+  if (action) {
+    valve->openValve();
+  } else {
+    valve->closeValve();
+  }
+}
+
```
Comment (fazerlicourice7) [inline Brain/brain_utils.h:137]: wtf. was this not in already
:P my bad.

**Review by fazerlicourice7 — CHANGES_REQUESTED** [review]
tiny change

File: Brain/brain_utils.h (hunk 2 above)
Comment (vainaviv) [reply Brain/brain_utils.h:137]: Haha no worries :)

File: Brain/Brain.ino (hunk 1 above)
Comment (vainaviv) [reply Brain/Brain.ino:45]: Again why do we need numSensors I don't see it in use

Comment (fazerlicourice7) [reply Brain/Brain.ino:45]: scratch that. my b

**Review by fazerlicourice7 — APPROVED** [review]

**Pull request merged**